src/client/eppo-client.ts

```diff
@@ -488,72 +497,82 @@ export default class EppoClient {
       )?.key;
 
       if (banditKey) {
-        // If no actions are passed, return the default value and do not log a bandit assignment
-        if (!Object.keys(actions).length) {
-          return {
-            ...defaultResult,
-            evaluationDetails: flagEvaluationDetailsBuilder.buildForNoneResult(
-              'NO_ACTIONS_SUPPLIED_FOR_BANDIT',
-              'No actions passed for a flag that evaluated to a bandit variation',
-            ),
-          };
-        }
-
-        // Retrieve the model parameters for the bandit
-        const banditParameters = this.banditModelConfigurationStore?.get(banditKey);
-
-        if (!banditParameters) {
-          throw new Error('No model parameters for bandit ' + banditKey);
-        }
-
-        const banditModelData = banditParameters.modelData;
-        const contextualSubjectAttributes =
-          this.ensureContextualSubjectAttributes(subjectAttributes);
-        const actionsWithContextualAttributes = this.ensureActionsWithContextualAttributes(actions);
-        const banditEvaluation = this.banditEvaluator.evaluateBandit(
+        evaluationDetails.banditKey = banditKey;
+        action = this.evaluateBanditAction(
```

Comment:
moved bandit evaluation into its own method that returns the selected action to clean up this top-level method